security_best_practices/security_best_practices.sp

```diff
@@ -0,0 +1,243 @@
+      case
+        when key_expiry_disabled then name || ' has key expiration disabled.'
+        else name || ' has key expiration enabled.'
+      end as reason,
+      tailnet_name
+    from
+      tailscale_device;
+  EOT
+}
+
+control "tailscale_device_network_boundary_protected" {
+  title       = "Protect your network boundary"
+  description = "Restrict access to your private network, e.g., using a firewall. Tailscale allows you to easily connect your devices no matter their local area network, and ensures that traffic between your devices is end-to-end encrypted."
+  sql = <<-EOT
+    select
+      id as resource,
+      case
+        when blocks_incoming_connections then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when blocks_incoming_connections then name || ' access restricted to private network.'
+        else name || ' access not restricted to private network.'
+      end as reason,
+      tailnet_name
+    from
+      tailscale_device;
+  EOT
+}
+
+control "tailscale_device_upgrade_clients_in_timely_manner" {
+  title       = "Upgrade Tailscale clients in a timely manner"
+  description = "Upgrade Tailscale clients regularly, in a timely manner. Tailscale frequently introduces new features and patches existing versions, including security patches."
+  sql  = <<-EOT
+    select
+      id as resource,
+      case
+        when not update_available then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when not update_available then name || ' is updated.'
+        else name || ' is not updated.'
+      end as reason,
+      tailnet_name
+    from
+      tailscale_device;
+  EOT
+}
+
+control "tailscale_tailnet_acl_groups_used" {
+  title       = "Use groups in ACLs"
+  description = "Using groups allows identities to be controlled based on job function. If someone leaves an organization or changes roles, you can adjust the group membership rather than update all of their ACLs."
+  sql = <<-EOT
+    select
+      tailnet_name as resource,
+      case
+        when acl_groups is not null then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when acl_groups is not null then tailnet_name || ' uses ACL group.'
+        else tailnet_name || ' does not use ACL group.'
+      end as reason
+    from
+      tailscale_tailnet;
+  EOT
+}
+
+control "tailscale_tailnet_acl_tags_used" {
+  title       = "Use tags in ACLs"
+  description = "Use tags to manage devices. Using tags allows you to define access to devices based on purpose, rather than based on owner."
+  sql = <<-EOT
+    select
+      tailnet_name as resource,
+      case
+        when acl_tag_owners is not null then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when acl_tag_owners is not null then tailnet_name || ' uses ACL tags.'
+        else tailnet_name || ' does not use ACL tags.'
+      end as reason
+    from
+      tailscale_tailnet;
+  EOT
+}
+
+control "tailscale_tailnet_key_unused" {
```

Comment:
Is this control checking the right thing? From what I understand, the guidance is to remove unused keys, but this is just checking for expired keys, which is a different check IMO.

If we can't get usage data for keys, I think we should remove this control instead of having one that checks something entirely different.